dockers/chart-releaser/main.go

```diff
@@ -62,20 +62,22 @@ func init() {
 }
 
 func main() {
-	ctx, cancel := context.WithTimeout(context.Background(), timeout)
-	defer cancel()
+	if err := run(); err != nil {
+		log.Fatalf("Failed to run: %s", err.Error())
+	}
+}
 
-	ch := make(chan os.Signal, 1)
-	signal.Notify(ch, syscall.SIGINT, syscall.SIGTERM)
-	defer signal.Stop(ch)
+func run() error {
+	pctx, cancel := context.WithTimeout(context.Background(), timeout)
+	defer cancel()
+	ctx, stop := signal.NotifyContext(pctx, syscall.SIGINT, syscall.SIGTERM, os.Interrupt)
+	defer stop()
 
 	go func() {
-		select {
-		case <-ch:
-			cancel()
-		case <-ctx.Done():
-			return
-		}
+		log.Println("Waiting for signals...")
+		<-ctx.Done()
+		log.Println("Received signal, exiting...")
+		stop()
 	}()
```

Comment:
By using NotifyContext I think we won't need the goroutine to wait for the context completion.

I think this is enough. How do you think?

``` golang
	ctx, cancel := context.WithTimeout(context.Background(), timeout)
	defer cancel()
	ctx, _  = signal.NotifyContext(pctx, syscall.SIGINT, syscall.SIGTERM, os.Interrupt)
```